### ISSUE TITLE
Couldn't  start daemon after upgrading gradle from 6.3 to 6.4 and above. Problem with binding to loopback address

### ISSUE THREAD

**Issue by magowy** [gradle/gradle#15343]
### Expected Behavior

Bind properly  to loopback network interface and start deamon. 

### Current Behavior

After upgrading gradle from 6.3 to 6.4  or above (including 6.8-rc1) we encountered build problem on our workstation. 
Problem includes importing and running build from command line and IDE (InteliJ).

When running build  **gradle clean war**  build fails.

In daemon log i found 

```
2020-11-27T16:07:06.256+0100 [DEBUG] [org.gradle.launcher.daemon.bootstrap.DaemonMain] Assuming the daemon was started with following jvm opts: [--add-opens, java.base/java.util=ALL-UNNAMED, --add-opens, java.base/java.lang=ALL-UNNAMED, --add-opens, java.base/java.lang.invoke=ALL-UNNAMED, --add-opens, java.prefs/java.util.prefs=ALL-UNNAMED, -XX:MaxMetaspaceSize=256m, -XX:+HeapDumpOnOutOfMemoryError, -Xms256m, -Xmx512m, -Dfile.encoding=windows-1250, -Duser.country=PL, -Duser.language=pl, -Duser.variant]
2020-11-27T16:07:06.357+0100 [DEBUG] [org.gradle.internal.nativeintegration.services.NativeServices] Native-platform posix files integration is not available. Continuing with fallback.
2020-11-27T16:07:06.371+0100 [INFO] [org.gradle.launcher.daemon.server.Daemon] start() called on daemon - DefaultDaemonContext[uid=8a28381e-b5e8-4f25-8a40-666f14c68c3d,javaHome=C:\Program Files\Java\zulujdk11.0.9.1,daemonRegistryDir=C:\Users\myUser\.gradle\daemon,pid=15616,idleTimeout=10800000,priority=NORMAL,daemonOpts=--add-opens,java.base/java.util=ALL-UNNAMED,--add-opens,java.base/java.lang=ALL-UNNAMED,--add-opens,java.base/java.lang.invoke=ALL-UNNAMED,--add-opens,java.prefs/java.util.prefs=ALL-UNNAMED,-XX:MaxMetaspaceSize=256m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=windows-1250,-Duser.country=PL,-Duser.language=pl,-Duser.variant]
2020-11-27T16:07:07.450+0100 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interface Software Loopback Interface 1
2020-11-27T16:07:07.453+0100 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? true
2020-11-27T16:07:07.457+0100 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Error while querying interface name:lo (Software Loopback Interface 1) for IP addresses
java.net.SocketException: Unrecognized Windows Sockets error: 11050: no further information
	at java.base/java.net.Inet6AddressImpl.isReachable0(Native Method)
	at java.base/java.net.Inet6AddressImpl.isReachable(Inet6AddressImpl.java:88)
	at java.base/java.net.InetAddress.isReachable(InetAddress.java:545)
	at java.base/java.net.InetAddress.isReachable(InetAddress.java:504)
	at org.gradle.internal.remote.internal.inet.InetAddresses.analyzeNetworkInterface(InetAddresses.java:63)
	at org.gradle.internal.remote.internal.inet.InetAddresses.analyzeNetworkInterfaces(InetAddresses.java:47)
	at org.gradle.internal.remote.internal.inet.InetAddresses.<init>(InetAddresses.java:40)
	at org.gradle.internal.remote.internal.inet.InetAddressFactory.init(InetAddressFactory.java:99)
	at org.gradle.internal.remote.internal.inet.InetAddressFactory.getLocalBindingAddress(InetAddressFactory.java:73)
	at org.gradle.internal.remote.internal.inet.TcpIncomingConnector.accept(TcpIncomingConnector.java:59)
	at org.gradle.launcher.daemon.server.DaemonTcpServerConnector.start(DaemonTcpServerConnector.java:86)
	at org.gradle.launcher.daemon.server.Daemon.start(Daemon.java:169)
	at org.gradle.launcher.daemon.bootstrap.DaemonMain.doAction(DaemonMain.java:125)
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:50)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:60)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:37)
	at org.gradle.launcher.daemon.bootstrap.GradleDaemon.main(GradleDaemon.java:22)
2020-11-27T16:07:07.459+0100 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interface Microsoft 6to4 Adapter
2020-11-27T16:07:07.463+0100 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
```
Omitted the logs for all the other interfaces as I have a few, but all of them were identified correctly as not being loopback interfaces.

After iterating over all interface there is a message:

```
2020-11-27T16:07:07.757+0100 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddressFactory] No loopback address for local binding, using fallback 0.0.0.0/0.0.0.0
2020-11-27T16:07:07.757+0100 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddressFactory] No loopback addresses for communication, using remote addresses instead.
2020-11-27T16:07:07.762+0100 [DEBUG] [org.gradle.internal.remote.internal.inet.TcpIncomingConnector] Listening on [b4fbbff4-cc3a-43e0-bf3f-52f64095fb65 port:18130, addresses:[0.0.0.0/0.0.0.0]].
2020-11-27T16:07:07.768+0100 [DEBUG] [org.gradle.launcher.daemon.server.Daemon] Daemon starting at: Fri Nov 27 16:07:07 CET 2020, with address: [b4fbbff4-cc3a-43e0-bf3f-52f64095fb65 port:18130, addresses:[0.0.0.0/0.0.0.0]]
2020-11-27T16:07:07.768+0100 [INFO] [org.gradle.launcher.daemon.server.DaemonRegistryUpdater] Advertising the daemon address to the clients: [b4fbbff4-cc3a-43e0-bf3f-52f64095fb65 port:18130, addresses:[0.0.0.0/0.0.0.0]]
2020-11-27T16:07:07.768+0100 [DEBUG] [org.gradle.launcher.daemon.server.DaemonRegistryUpdater] Advertised daemon context: DefaultDaemonContext[uid=8a28381e-b5e8-4f25-8a40-666f14c68c3d,javaHome=C:\Program Files\Java\zulujdk11.0.9.1,daemonRegistryDir=C:\Users\myUser\.gradle\daemon,pid=15616,idleTimeout=10800000,priority=NORMAL,daemonOpts=--add-opens,java.base/java.util=ALL-UNNAMED,--add-opens,java.base/java.lang=ALL-UNNAMED,--add-opens,java.base/java.lang.invoke=ALL-UNNAMED,--add-opens,java.prefs/java.util.prefs=ALL-UNNAMED,-XX:MaxMetaspaceSize=256m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=windows-1250,-Duser.country=PL,-Duser.language=pl,-Duser.variant]
2020-11-27T16:07:07.769+0100 [DEBUG] [org.gradle.launcher.daemon.registry.PersistentDaemonRegistry] Storing daemon address: [b4fbbff4-cc3a-43e0-bf3f-52f64095fb65 port:18130, addresses:[0.0.0.0/0.0.0.0]], context: DefaultDaemonContext[uid=8a28381e-b5e8-4f25-8a40-666f14c68c3d,javaHome=C:\Program Files\Java\zulujdk11.0.9.1,daemonRegistryDir=C:\Users\myUser\.gradle\daemon,pid=15616,idleTimeout=10800000,priority=NORMAL,daemonOpts=--add-opens,java.base/java.util=ALL-UNNAMED,--add-opens,java.base/java.lang=ALL-UNNAMED,--add-opens,java.base/java.lang.invoke=ALL-UNNAMED,--add-opens,java.prefs/java.util.prefs=ALL-UNNAMED,-XX:MaxMetaspaceSize=256m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=windows-1250,-Duser.country=PL,-Duser.language=pl,-Duser.variant]
```
It's is trying to bind to 0.0.0.0 which is forbidden on our company because  of VPN konfiguration. Only localhost/loopback binding is allowed. 
At the end the daemon fails to start with message (from reason above)

```
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] ----- End of the daemon log -----
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.daemon.client.DefaultDaemonConnector.connectToDaemonWithId(DefaultDaemonConnector.java:257)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.daemon.client.DefaultDaemonConnector.doStartDaemon(DefaultDaemonConnector.java:222)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.daemon.client.DefaultDaemonConnector.startDaemon(DefaultDaemonConnector.java:211)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.daemon.client.DefaultDaemonConnector.connect(DefaultDaemonConnector.java:133)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.daemon.client.DaemonClient.execute(DaemonClient.java:144)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.daemon.client.DaemonClient.execute(DaemonClient.java:98)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:57)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:212)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.DefaultCommandLineActionFactory$ParseAndBuildAction.execute(DefaultCommandLineActionFactory.java:275)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.DefaultCommandLineActionFactory$ParseAndBuildAction.execute(DefaultCommandLineActionFactory.java:247)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.DebugLoggerWarningAction.execute(DebugLoggerWarningAction.java:82)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.DebugLoggerWarningAction.execute(DebugLoggerWarningAction.java:30)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.WelcomeMessageAction.execute(WelcomeMessageAction.java:92)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.WelcomeMessageAction.execute(WelcomeMessageAction.java:38)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.NativeServicesInitializingAction.execute(NativeServicesInitializingAction.java:44)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.NativeServicesInitializingAction.execute(NativeServicesInitializingAction.java:26)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:41)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:26)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.cli.DefaultCommandLineActionFactory$WithLogging.execute(DefaultCommandLineActionFactory.java:240)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.Main.doAction(Main.java:35)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:50)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:60)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:37)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.GradleMain.main(GradleMain.java:31)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: org.gradle.internal.remote.internal.ConnectException: Could not connect to server [b4fbbff4-cc3a-43e0-bf3f-52f64095fb65 port:18130, addresses:[/0.0.0.0]]. Tried addresses: [/0.0.0.0].
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.remote.internal.inet.TcpOutgoingConnector.connect(TcpOutgoingConnector.java:67)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.daemon.client.DefaultDaemonConnector.connectToDaemon(DefaultDaemonConnector.java:269)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.launcher.daemon.client.DefaultDaemonConnector.connectToDaemonWithId(DefaultDaemonConnector.java:255)
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 27 more
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: java.net.SocketException: Permission denied: no further information
2020-11-27T16:07:07.873+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
2020-11-27T16:07:07.874+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:776)
2020-11-27T16:07:07.874+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at java.base/sun.nio.ch.SocketAdaptor.connect(SocketAdaptor.java:120)
2020-11-27T16:07:07.874+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.remote.internal.inet.TcpOutgoingConnector.tryConnect(TcpOutgoingConnector.java:81)
2020-11-27T16:07:07.874+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.remote.internal.inet.TcpOutgoingConnector.connect(TcpOutgoingConnector.java:54)
2020-11-27T16:07:07.874+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 29 more
2020-11-27T16:07:07.874+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
2020-11-27T16:07:07.874+0100 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
```

Adding -Djava.net.preferIPv4Stack=true  to build command din't help. Looks like gradle daemon is ignoring this parameter. 
I also tried to disable IPv6 protocol on all my network adapter, but it didn't help either. 

### Context

The daemon could not started and we are unable to build application from command line o import project to InteliJ with gradle  6.4 and above.  There is no problem with gradle version 6.0.1 to 6.3

### Steps to Reproduce 

It seams be replicable in vpn connection  with no privilege to bind  ports on interface other than looppack. 

### Your Environment
Windows 10   version 10.0.19041 build 19041
Running on VPN with lockdown mode (application  are not allowed to bind to IP's other than loopback/localhost )
VPN Agent PulseSecure

> Lock down mode is designed to prohibit network communication outside of the VPN Tunnel

**Not working build** 

>gradle -v

------------------------------------------------------------
Gradle 6.7.1
------------------------------------------------------------

Build time:   2020-11-16 17:09:24 UTC
Revision:     2972ff02f3210d2ceed2f1ea880f026acfbab5c0

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          11.0.9.1 (Azul Systems, Inc. 11.0.9.1+1-LTS)
OS:           Windows 10 10.0 amd64

> gradle -v

**Working build** 
Gradle 6.3

Build time:   2020-03-24 19:52:07 UTC
Revision:     bacd40b727b0130eeac8855ae3f9fd9a0b207c60

Kotlin:       1.3.70
Groovy:       2.5.10
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          11.0.9.1 (Azul Systems, Inc. 11.0.9.1+1-LTS)
OS:           Windows 10 10.0 amd64

```

**Comment by holgerbrandl** [comment]
Is there a workaround?

**Comment by JackTiger** [comment]
@holgerbrandl  By the way, Can Gradle 6.3 support disabling the daemon by default?

I find after I set "org.gradle.daemon=true" in gradle.properties file, the gradle daemon still started。

**Comment by magowy** [comment]
> Is there a workaround?

I tried several thing but unfortunately i I didn't find simple warkaround. There is almost no information about this type of exception.

What i did was to build my own version of gradle using org.gradle.internal.remote.internal.inet.InetAddresses  from earlier version of gradle (from gradle 6.3)

**Comment by mlopatkin** [comment]
This issue should be fixed in 7.0 by 83f43237865ce216bdfc4204a9cb99e6eaaa9616. Please check with the recent Gradle version. Feel free to reply if the issue is still reproducible, I'll reopen it then.

>Can Gradle 6.3 support disabling the daemon by default?

You can run without the daemon but this is a complicated task and this mode is being phased out, so I suggest not to pursue that.
